src/CalcViewModel/Common/TraceLogger.cpp

```diff
@@ -236,782 +121,215 @@ namespace CalculatorApp
+            buttonId == (int)NumbersAndOperatorsEnum::IsStandardMode || buttonId == (int)NumbersAndOperatorsEnum::None)
         {
-            index = findIndex[index];
-            return true;
-        }
-        return false;
-    }
-
-    void TraceLogger::UpdateWindowCount(size_t windowCount)
-    {
-        maxWindowCount = (maxWindowCount > windowCount) ? maxWindowCount : windowCount;
-        windowLaunchCount++;
-    }
-
-    void TraceLogger::LogMaxWindowCount()
-    {
-        if (!GetTraceLoggingProviderEnabled())
-            return;
-
-        LoggingFields fields{};
-        fields.AddUInt32(L"WindowCount", (unsigned int)maxWindowCount);
-        LogTelemetryEvent(EVENT_NAME_MAX_WINDOW_COUNT, fields);
-    }
-
-    void TraceLogger::LogWindowActivated() const
-    {
-        if (!GetTraceLoggingProviderEnabled())
             return;
-
-        LoggingFields fields{};
-        LogTelemetryEvent(EVENT_NAME_WINDOW_LAUNCHED_PROTOCOL, fields);
-    }
-
-    void TraceLogger::LogWindowLaunched() const
-    {
-        if (!GetTraceLoggingProviderEnabled())
-            return;
-
-        LoggingFields fields{};
-        LogTelemetryEvent(EVENT_NAME_WINDOW_LAUNCHED_TILESEARCH, fields);
-    }
-
-    void TraceLogger::LogBitLengthButtonUsed(int windowId)
-    {
-        if (bitLengthButtonUsage.find(windowId) == bitLengthButtonUsage.end())
-        {
-            bitLengthButtonUsage.insert(pair<int, int>(windowId, 1));
         }
-        else
-        {
-            bitLengthButtonUsage[windowId]++;
-        }
-        if ((bitLengthButtonUsage[windowId] == 5) && !bitLengthButtonLoggedInSession)
-        {
-            LoggingFields fields{};
-            fields.AddUInt32(L"WindowId", windowId);
-            LogTelemetryEvent(EVENT_NAME_BITLENGTH_BUTTON_USED, fields);
-
-            bitLengthButtonLoggedInSession = true;
-        }
-    }
 
-    void TraceLogger::LogRadixButtonUsed(int windowId)
-    {
-        if (radixButtonUsage.find(windowId) == radixButtonUsage.end())
+        // Writer lock for the static resources
+        reader_writer_lock::scoped_lock lock(s_traceLoggerLock);
+        vector<ButtonLog>::iterator it = std::find_if(
+            buttonLog.begin(), buttonLog.end(), [buttonId, mode](const ButtonLog& bLog) -> bool { return bLog.buttonId == buttonId && bLog.mode == mode; });
+        if (it != buttonLog.end())
         {
-            radixButtonUsage.insert(pair<int, int>(windowId, 1));
+            it->count++;
         }
         else
         {
-            radixButtonUsage[windowId]++;
-        }
-        if ((radixButtonUsage[windowId] == 2) && !radixButtonLoggedInSession)
-        {
-            LoggingFields fields{};
-            fields.AddUInt32(L"WindowId", windowId);
-            LogTelemetryEvent(EVENT_NAME_RADIX_BUTTON_USED, fields);
-
-            radixButtonLoggedInSession = true;
+            NumbersAndOperatorsEnum button = safe_cast<NumbersAndOperatorsEnum>(buttonId);
```

Comment:
>safe_cast<NumbersAndOperatorsEnum>(buttonId); [](start = 45, length = 45)

safe_cast is a CX-specific weird thing--in this case you want static_cast

Also, can we just use NumbersAndOperatorsEnum everywhere until the point in LogButtonUsage where we actually need to stick this in a tracelogging field